Source/JavaScriptCore/runtime/ClonedArguments.cpp

```diff
@@ -40,24 +40,26 @@ ClonedArguments::ClonedArguments(VM& vm, Structure* structure, Butterfly* butter
 {
 }
 
-ClonedArguments* ClonedArguments::createEmpty(
-    VM& vm, Structure* structure, JSFunction* callee, unsigned length)
+ClonedArguments* ClonedArguments::createEmpty(VM& vm, Structure* structure, JSFunction* callee, unsigned length, Butterfly* butterfly)
 {
     unsigned vectorLength = length;
     if (vectorLength > MAX_STORAGE_VECTOR_LENGTH)
         return nullptr;
 
-    Butterfly* butterfly;
     if (UNLIKELY(structure->mayInterceptIndexedAccesses() || structure->storedPrototypeObject()->needsSlowPutIndexing())) {
-        butterfly = createArrayStorageButterfly(vm, nullptr, structure, length, vectorLength);
-        butterfly->arrayStorage()->m_numValuesInVector = vectorLength;
+        if (!butterfly) {
+            butterfly = createArrayStorageButterfly(vm, nullptr, structure, length, vectorLength);
+            butterfly->arrayStorage()->m_numValuesInVector = vectorLength;
+        }
     } else {
-        IndexingHeader indexingHeader;
-        indexingHeader.setVectorLength(vectorLength);
-        indexingHeader.setPublicLength(length);
-        butterfly = Butterfly::tryCreate(vm, nullptr, 0, structure->outOfLineCapacity(), true, indexingHeader, vectorLength * sizeof(EncodedJSValue));
-        if (!butterfly)
-            return nullptr;
+        if (!butterfly) {
+            IndexingHeader indexingHeader;
+            indexingHeader.setVectorLength(vectorLength);
+            indexingHeader.setPublicLength(length);
+            butterfly = Butterfly::tryCreate(vm, nullptr, 0, structure->outOfLineCapacity(), true, indexingHeader, vectorLength * sizeof(EncodedJSValue));
+            if (!butterfly)
+                return nullptr;
```

Comment:
Should we ever check and return `nullptr` here instead of a RELEASE_ASSERT?  It seems the client below assumes that it never fails to allocate.  I guess it's ok to leave as is for now, but I suspect we should change this to a RELEASE_ASSERT later.

Reply:
For now, I'm not changing it since it is the same to the current behavior.